snap/info.go

```diff
@@ -1001,6 +1003,24 @@ func ReadInfo(name string, si *SideInfo) (*Info, error) {
 		return nil, &invalidMetaError{Snap: name, Revision: si.Revision, Msg: err.Error()}
 	}
 
+	err = addImplicitHooks(info)
+	if err != nil {
+		return nil, &invalidMetaError{Snap: name, Revision: si.Revision, Msg: err.Error()}
+	}
+
+	_, instanceKey := SplitInstanceName(name)
+	info.InstanceKey = instanceKey
+
+	return info, nil
+}
+
+// ReadInfo reads the snap information for the installed snap with the given name and given side-info.
+func ReadInfo(name string, si *SideInfo) (*Info, error) {
+	info, err := ReadInfoExceptSize(name, si)
+	if err != nil {
+		return nil, err
+	}
+
 	mountFile := MountFile(name, si.Revision)
 	st, err := os.Stat(mountFile)
 	if os.IsNotExist(err) {
```

Comment:
Am I being silly, or is the only reason this works today because apparmor leaks stats? Consider: if the snap I'm `try`ing doesn't use the `home` plug, but the source code for the snap is in `~/src/snaps/` then the symlink that is `mountFile` is pointing to a location inaccessible by snap-exec. However, because appamor leaks stats, we happen to have the ability to determine that the symlink is pointing to something that exists.

Reply:
Apparmor doesn't check stat, that's correct. As for Lstat - perhaps, cc @chipaca for opinion.

In any case inside the mount namespace we cannot even attempt to follow the symlink as someone may craft it to be malicious.